### ISSUE TITLE
Skyrim Remastered Support

### ISSUE THREAD

**Issue by Mardoxx** [mlheur/SKSE-SE#2]
This doesnt work with Skyrim Remastered.

**Comment by mlheur** [comment]
No, but I have so far I managed to get it to compile under x64 platform and changed most 32 bit memory address variables to platform specific data types in the common, skse_loader and steam_loader components.  I still need to do this for the actual skse dll but needed to wait to see if game data was going to have 64 bit types, like FormIDs.

Then, once that's all done there's still the task of finding where all the hooks have moved to in SkyrimSE.exe compared to where they were in TESV.exe.  I'm still not sure how I'm going to accomplish that.

**Comment by Mardoxx** [comment]
Absolute genious..

**Comment by GregJM** [comment]
Do you have the assembly or binary code for the hook routines?

**Comment by EdwardBlair** [comment]
How do you expect to do inline ASM with a 64bit target?

**Comment by mlheur** [comment]
@GregJM - no, i'm not even sure if Bethesda has used the same NetImmerse or ScaleForm libraries compiled into the new EXE, so all the hooks in the original SKSE could be totally invalid.

@EdwardBlair - As far as I've seen the amount of __asm is minimal compared to where SKSE is writing actual binary opcodes as data overtop of the mmapped EXE loaded in RAM.  There are a few places already where things like the Instruction Pointer register was referred to as Eip and I had to change that to Rip just for x64 platform compilation.

In short, I'm feeling around blindly and keeping my fingers crossed.  I had tried contacting the SKSE team at silverlight a few times in the past about the bugs and fixes I already applied for TESV and never got any response.  I decided to do this for myself as a learning experience and in case SKSE has been abandoned.  So far I haven't been able to find any word that the team at silverlight is even interested in working on SSE.

This is on github mostly so I can roll back any major code-breaking bonehead changes and to have an offsite copy from my workstation.  I'd much rather be a contributor on the grunt work than a prime developer, but until I see something more concrete I'm going to keep pushing forward.

**Comment by Eperty123** [comment]
Hey mlheur,

So what exactly needs to be done? And how can we get the addresses of the the hooks?

**Comment by viperfan7** [comment]
I'm wondering if any code from F4SE could be ported over to SKSE-SE, seeing as its the same engine now.

May simplify things

**Comment by and00r** [comment]
this is such a load of BS

only reason why skse is delayed is because of precious consoles and their money. 
skyrim shouldn't even need skse in the first place. 

todd is going to die if one of his products stops selling without 2000% exclusive rights to him in triplicate copyright and super serious EULAs which are not actual legal binding contracts but idiots keep attaching them to software anyway

**Comment by Tonycubed2** [comment]
@Mardoxx : thank you for trying. All my major mods use skse. I can't code, but if I can send you a 12 pack of Red Bull's or something let me know.  I have extra hardware laying around if you need PC components , etc.

**Comment by mlheur** [comment]
@Eperty123 - I'm not sure, if I could answer that I would probably need less help...
@viperfan7  - you deserve a gold star.  I've never played any of the Fallout games so it never even crossed my mind, but yeah by studying that code (and other __SE) will be a big help.  Now I just need to get my hands on Fallout4.exe v 1.7.22.*  At least now I know the __asm problem was solved in f4se by using Xbyaak JIT 

Based on what I've seen so far it looks to me like skse and f4se were built with good old-school reverse engineering, using decompilers and memory watchers, rather than having intimate knowledge of how the engine was compiled.  Right now I'm using x64dbg on SkyrimSE.exe and x32dbg on TESV.exe to see if I can compare and find the same signatures in the opcodes.

The good news is that with the f4se code base I'm at least able to confirm that the staticly linked libraries are the same or similar from TESV to Fallout4.exe.  What's confusing to me though is that f4se is built for x64 platform but a lot of the magic number addresses I'm seeing in the main dll are still only 32-bit.

**Comment by Mardoxx** [comment]
The exe isn't, eh, in a format, so to speak, that can be dealt with nicely. As soon as it is work can begin. No point in looking at it until that time. 

Not sure if you'vre aware @mlheur, but skse while it is open source, it still remains all rights reserved. So hosting it here and plastering MIT license on it is not only shady but will just provoke people.

**Comment by mlheur** [comment]
Thanks for the head's up - I thought I saw the MIT license somewhere else in the source base but I can't find it now.  That could've been from bsaopt or xEdit since I was bouncing around from a few code bases.

I'm happy to delete the repo, if the silverlight team wants me to.  I was going to make it private but I'm too cheap to pay for github.
